addons/web_editor/static/src/js/wysiwyg_snippets/snippets.options.js

```diff
@@ -834,6 +834,36 @@ registry.background = SnippetOption.extend({
             this.$target.removeClass('oe_img_bg oe_custom_bg');
         }
     },
+    /**
+     * Handles a background video change.
+     *
+     * @param {object} data
+     */
+    backgroundVideo: function (previewMode, data) {
```

Comment:
You define this method as if it is an option which can be used with `data-background-video="..."` except that nobody will be able to use this because your data seems to require an inner iframe etc. You don't need all of this, and you should not have to parse the data to add custom parameters in the url, etc. Why do you need to handle "playlist" here ?

Reply:
@var-odoo 

(Reply of message in odoo.com chatter)
> I don't understand "You don't need all of this, and you should not have to parse the data to add custom parameters in the url, etc." on PR of this task can you please explain what changes I have to do?

This method defines an option for handling `data-background-video="..."` in snippets options, there will never be ones. At best, all you do here should be done in `_setCustomVideo` directly. But I also don't understand why you need to parse and search the URL to add "playlist" in it ? It should already be in there if necessary (handled by the media dialog)... why doing it here ?

Reply:
Hello @qsm-odoo please check this commit [https://github.com/odoo/odoo/commit/b9cd7318e3506b92a0f04b59dc37cf098bb2e01b](https://github.com/odoo/odoo/commit/b9cd7318e3506b92a0f04b59dc37cf098bb2e01b) I have removed the `backgroundVideo` method and from now the video will directly set from the` _setCustomVideo` method if it's good then I will squash this commit. Also, the previewMode is used for binding the color event with background video so when we hover on the color box the video will be removed and if color is not selected then the video will be reset same as we are doing with the background image you can check this video [Video](https://drive.google.com/open?id=1218mMXtb5ziwEw8L-UgZzUfk3yX4wti8). 'Playlist' is used for looping in media block snippet so according to it I have also used but it is not required now so I will remove it. Thanks :)

Reply:
Ok, I'll check again when all changes have been done